File: apps/api/schema/queries.py
import graphene

from models import Crew, Job, db_session

from .types import BoardType


class Query(graphene.ObjectType):
    board = graphene.Field(
        BoardType,
        dates=graphene.List(graphene.NonNull(graphene.Date), required=True),
        required=True,
    )

    def resolve_board(self, info, dates):
        crews = db_session.query(Crew).all()
        jobs = db_session.query(Job).filter(Job.date.in_(dates)).all()
        return BoardType(crews=crews, jobs=jobs)

File: apps/web/src/scheduler/data/queries.ts
import { gql } from "@apollo/client"
import type { Crew, Job } from "../core/types"

export const BOARD_QUERY = gql`
  query Board($dates: [Date!]!) {
    board(dates: $dates) {
      crews {
        id
        name
      }
      jobs {
        id
        crewId
        title
        date
        startTime
        durationMinutes
      }
    }
  }
`

export type BoardQueryData = {
  board: {
    crews: Crew[]
    jobs: Job[]
  }
}

export type BoardQueryVariables = {
  dates: string[]
}
